assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,338 @@
+        type=str,
+        help="Name of the endpoint",
+    )
+    parser.add_argument("--deployment_name", type=str, help="Name of the the deployment")
+    parser.add_argument(
+        "--compute_name",
+        type=str,
+        help="Name of the compute target to execute the batch scoring jobs on",
+    )
+    parser.add_argument(
+        "--size",
+        type=str,
+        help="Compute instance size to deploy model",
+        default="Standard_NC24s_v3",
+    )
+    parser.add_argument(
+        "--min_instances",
+        type=int,
+        default=0,
+        help="Minimum number of instances of the compute cluster",
+    )
+    parser.add_argument(
+        "--max_instances",
+        type=int,
+        default=1,
+        help="Maximum number of instances of the compute cluster",
+    )
+    parser.add_argument(
+        "--idle_time_before_scale_down",
+        type=int,
+        default=120,  # 2min
+        help="Node Idle Time before scaling down amlCompute",
+    )
+    parser.add_argument(
+        "--output_file_name",
+        type=str,
+        default="predictions.csv",
+        help="Name of the batch scoring output file",
+    )
+    parser.add_argument(
+        "--max_concurrency_per_instance",
+        type=int,
+        default=1,
+        help="The maximum number of parallel scoring_script runs per instance",
+    )
+    parser.add_argument(
+        "--error_threshold",
+        type=int,
+        default=-1,
+        help="The number of file failures that should be ignored",
+    )
+    parser.add_argument(
+        "--max_retries",
+        type=int,
+        default=3,
+        help="The maximum number of retries for a failed or timed-out mini batch",
+    )
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=500,  # 500sec
+        help="The timeout in seconds for scoring a single mini batch.",
+    )
+    parser.add_argument(
+        "--logging_level",
+        type=str,
+        default="info",
+        help="The log verbosity level",
+    )
+    parser.add_argument(
+        "--mini_batch_size",
+        type=int,
+        default=10,
+        help="The number of files the code_configuration.scoring_script can process in one run() call",
+    )
+    parser.add_argument(
+        "--instance_count",
+        type=int,
+        help="The number of nodes to use for each batch scoring job",
+        default=1,
+        choices=range(1, MAX_INSTANCE_COUNT),
+    )
+    parser.add_argument(
+        "--model_deployment_details",
+        type=str,
+        help="Json file to which deployment details will be written",
+    )
+    # parse args
+    args = parser.parse_args()
+    logger.info(f"Args received {args}")
+    print("args received ", args)
+
+    return args
+
+
+def create_endpoint_and_deployment(ml_client, model_id, endpoint_name, deployment_name, args):
+    """Create endpoint and deployment and return details."""
+    if args.compute_name:
```

Comment:
nit: modularize more? Can me separated to say get_compute_name()?

Reply:
Done